### PR TITLE
SP-6255 / Step testing helpers

### PR DESCRIPTION
**Pull request by kostyay**

### CODE REVIEW

File: stesting/container/container.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+	if images.ID != "" {
+		return nil
+	}
+
+	var pull io.ReadCloser
+	err = backoff.Retry(func() error {
+		pull, err = cli.ImagePull(ctx, imageName, types.ImagePullOptions{})
+		if err != nil {
+			if _, ok := err.(errdefs.ErrNotFound); ok {
```
Comment (vic3lord) [inline stesting/container/container.go:49]:
```suggestion
			if errors.Is(err, errdefs.ErrNotFound) {
```

File: stesting/container/container.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
+package container
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
```
Comment (vic3lord) [inline stesting/container/container.go:9]:
```suggestion
	"strings"
	"errors"
```

File: stesting/container/container.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+func ParseStepOutput(stepOutput string) (map[string]string, error) {
+	if !strings.Contains(stepOutput, env.EndMarkerDefault) {
+		return nil, fmt.Errorf("failed to parse step output, no end marker found in output")
+	}
+
+	parsedOutput := strings.Split(stepOutput, env.EndMarkerDefault)
+
+	jsonOutput := parsedOutput[1]
+
+	var unmarshalJsonOutput map[string]string
```
Comment (vic3lord) [inline stesting/container/container.go:137]: Will the JSON always be just string to string? if not it should be `map[string]interface{}`

File: stesting/container/container.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+	err := json.Unmarshal([]byte(jsonOutput), &unmarshalJsonOutput)
+
+	return unmarshalJsonOutput, err
```
Comment (vic3lord) [inline stesting/container/container.go:141]:
```suggestion
	return unmarshalJSONOutput, json.Unmarshal([]byte(jsonOutput), &unmarshalJSONOutput)
```

**Comment by kostyay** [comment]
1. Fixed PR comments.
2. Added gabs as JSON parser
3. Added reviewdog and fixed linting errors

**Review by vic3lord — APPROVED** [review]

**Pull request merged**